

# Lab 3 Từ Yêu cầu đến Thu thập dữ liệu

Thời lượng: 120 phút

## Mục tiêu

Sau khi hoàn thành lab này, bạn sẽ có thể:

-   Hiểu về yêu cầu dữ liệu
-   Khám phá các bước của thu thập dữ liệu


## Mục lục

<div class="alert alert-block alert-info" style="margin-top: 20px">
    
1. [Yêu cầu Dữ liệu](#0)<br>
2. [Thu thập Dữ liệu](#2)<br>
</div>
<hr>


# Yêu cầu Dữ liệu <a id="0"></a>


<img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DS0103EN/labs/images/lab2_fig1_flowchart_data_requirements.png" width=500>


Trong các video bài giảng, chúng ta đã biết rằng phương pháp tiếp cận phân tích đã chọn sẽ xác định các yêu cầu về dữ liệu. Cụ thể, các phương pháp phân tích được sử dụng sẽ yêu cầu nội dung dữ liệu, kiểu định dạng và cách biểu diễn dữ liệu nhất định, được định hướng bởi kiến thức nền tảng về nghiệp vụ, lĩnh vực liên quan (domain knowledge).


Trong lab **Từ vấn đề đến phương pháp tiếp cận**, chúng ta đã xác định rằng việc tự động hóa quy trình xác định nguồn gốc ẩm thực của một công thức hoặc món ăn nhất định có thể thực hiện được bằng cách sử dụng các thành phần nguyên liệu của công thức hoặc món ăn đó. Để xây dựng một mô hình, chúng ta cần nguồn dữ liệu phong phú về các nền ẩm thực và công thức nấu ăn khác nhau.


Việc xác định dữ liệu cần thiết đáp ứng giai đoạn yêu cầu dữ liệu của phương pháp luận khoa học dữ liệu.

* * *


# Thu thập Dữ liệu <a id="2"></a>


<img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DS0103EN/labs/images/lab2_fig2_flowchart_data_collection.png" width=500>


Trong giai đoạn thu thập dữ liệu ban đầu, các nhà khoa học dữ liệu xác định và thu thập các nguồn dữ liệu sẵn có. Chúng có thể ở dạng dữ liệu có cấu trúc, phi cấu trúc và thậm chí là bán cấu trúc có liên quan đến nghiệp vụ, lĩnh vực của vấn đề/bài toán cần giải quyết (problem domain).


#### Web tổng hợp các công thức nấu ăn trực tuyến

Một nhà nghiên cứu tên Yong-Yeol Ahn đã tìm kiếm hàng chục nghìn công thức nấu ăn (món ăn và nguyên liệu) từ ba trang web khác nhau, đó là:


<img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DS0103EN/labs/images/lab2_fig3_allrecipes.png" width=500>

[www.allrecipes.com](http://www.allrecipes.com?cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

<img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DS0103EN/labs/images/lab2_fig4_epicurious.png" width=500>

[www.epicurious.com](http://www.epicurious.com?cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

<img src="https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DS0103EN/labs/images/lab2_fig5_menupan.png" width=500>

[www.menupan.com](http://www.menupan.com?cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


Để biết thêm thông tin về Yong-Yeol Ahn và nghiên cứu của anh ấy, bạn có thể đọc bài báo của anh ấy trên [Flavor Network and the Principles of Food Pairing](http://yongyeol.com/papers/ahn-flavornet-2011.pdf?cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DS0103EN-SkillsNetwork-20083987&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


May mắn là chúng ta sẽ không cần phải thực hiện bất kỳ việc thu thập dữ liệu nào vì dữ liệu mà chúng ta cần để đáp ứng mục tiêu được xác định trong giai đoạn hiểu về nghiệp vụ (business understanding) đã sẵn sàng.


Chúng ta đã có được dữ liệu và đặt nó ở máy chủ IBM. Hãy tải dữ liệu xuống và xem.


<strong> Lưu ý quan trọng: </strong> Xin lưu ý rằng bạn không cần biết cách lập trình trong python. Đoạn code sau đây nhằm minh họa giai đoạn thu thập dữ liệu, vì vậy sẽ không có vấn đề gì nếu bạn không hiểu từng dòng code riêng lẻ. Chứng chỉ này sẽ có một khóa học đầy đủ về lập trình bằng python, <a href="http://cocl.us/PY0101EN_DS0103EN_LAB2_PYTHON_edX"> Python cho Khoa học Dữ liệu </a>, khóa học này sẽ dạy bạn cách lập trình bằng Python.


### Sử dụng notebook này:

Để chạy bất kỳ cell code nào sau đây, bạn có thể nhập **Shift + Enter** để thực thi code trong một cell.

Cài đặt phiên bản mới của Python


In [1]:
# check Python version
!python -V

Python 3.10.5


Đọc dữ liệu từ máy chủ IBM vào khung dữ liệu _pandas_.


In [3]:
import pandas as pd # download library to read data into dataframe
pd.set_option('display.max_columns', None)

# Điền code vào đây
recipes = pd.read_csv("https://s3-api.us-geo.objectstorage.softlayer.net/cf-courses-data/CognitiveClass/DS0103EN/labs/data/recipes.csv")

print("Data read into dataframe!") # takes about 30 seconds

Data read into dataframe!


Nhấp đúp vào __đây__ để xem lời giải.
<!--
pd.read_csv
-->

Hiển thị một số hàng đầu tiên.


In [4]:
# Điền code vào đây
print(recipes.head())

      country almond angelica anise anise_seed apple apple_brandy apricot  \
0  Vietnamese     No       No    No         No    No           No      No   
1  Vietnamese     No       No    No         No    No           No      No   
2  Vietnamese     No       No    No         No    No           No      No   
3  Vietnamese     No       No    No         No    No           No      No   
4  Vietnamese     No       No    No         No    No           No      No   

  armagnac artemisia artichoke asparagus avocado bacon baked_potato balm  \
0       No        No        No        No      No    No           No   No   
1       No        No        No        No      No    No           No   No   
2       No        No        No        No      No    No           No   No   
3       No        No        No        No      No    No           No   No   
4       No        No        No        No      No    No           No   No   

  banana barley bartlett_pear basil bay bean beech beef beef_broth beef_liver  \

Nhấp đúp vào __đây__ để xem lời giải.
<!--
recipes.head()
-->

Nhận chiều của khung dữ liệu.


In [6]:
# Điền code vào đây
recipes.shape

(57691, 384)

Nhấp đúp vào __đây__ để xem lời giải.
<!--
recipes.shape
-->

Vậy, tập dữ liệu của chúng ta gồm 57.691 công thức nấu ăn. Mỗi hàng đại diện cho một công thức và đối với mỗi công thức, món ăn tương ứng được ghi lại cũng như xem 384 thành phần nguyên liệu có tồn tại trong công thức hay không, bắt đầu bằng hạnh nhân (almond) và kết thúc bằng bí ngòi (zucchini).

* * *


Bây giờ giai đoạn thu thập dữ liệu đã hoàn tất, các nhà khoa học dữ liệu thường sử dụng thống kê mô tả và kỹ thuật trực quan hóa để hiểu rõ hơn về dữ liệu và làm quen với nó. Về cơ bản, các nhà khoa học dữ liệu khám phá dữ liệu để:

- hiểu nội dung của nó,
- đánh giá chất lượng của nó,
- khám phá bất kỳ thông tin chi tiết sơ bộ thú vị nào, và
- xác định xem dữ liệu bổ sung có cần thiết để lấp bất kỳ khoảng trống nào trong dữ liệu hay không.


### Nguồn: IBM
